app/src/main/java/com/github/se/polyfit/viewmodel/meal/OverviewViewModel.kt

```diff
@@ -208,4 +211,13 @@ constructor(
       }
     }
   }
+
+  fun getCaloriesPerMealOccasionTodayLiveData(): LiveData<List<Pair<MealOccasion, Double>>> {
+
+    return localDataProcessor.getCaloriesPerMealOccasionTodayLiveData()
+  }
+
```

Comment:
so you could do something like
```
private val _caloriesPerMealOccasions = MediatorLiveData<List<Pair<MealOccasion, Double>>>
// This creates a readOnly LiveData variable
private val caloriesPerMealOccasions : LiveData<List<Pair<MealOccasion, Double>>> = caloriesPerMealOccasions

```

And then set the value of `caloriesPerMealOccasions` to `getCaloriesPerMealOccasionTodayLiveData()`

either in the viewModel or with something like this in the overviewScreen
```
mealSummary by viewModel.caloriesPerMealOccasions.observeAsStata(viewModel.getCaloriesPerMealOccasionTodayLiveData())
```

To keep the separation between the viewModel and the screen

Reply:
That is a great idea and that is basically what is implemented. The live data created in the local database and then it is passed to the overview screen where it is observed.